include/nbl/video/alloc/SimpleGPUBufferAllocator.h

```diff
@@ -31,7 +31,7 @@ namespace video
         public:
             typedef IGPUBuffer* value_type;
 
-            SimpleGPUBufferAllocator(IDriver* inDriver, const IDriverMemoryBacked::SDriverMemoryRequirements& bufferReqs) :
+            SimpleGPUBufferAllocator(ILogicalDevice* inDriver, const IDriverMemoryBacked::SDriverMemoryRequirements& bufferReqs) :
                             GPUMemoryAllocatorBase(inDriver), mBufferMemReqs(bufferReqs)
             {
             }
```

Comment:
we should get rid of `IGPUBuffer::pseudoMoveAssign` and just make the engine work with the fact that `SimpleGPUBufferAllocator::reallocate` can change the allocation completely